src/amd/compiler/aco_assembler.cpp

```diff
@@ -21,6 +21,11 @@ struct asm_context {
 void emit_instruction(asm_context& ctx, std::vector<uint32_t>& out, Instruction* instr)
 {
    uint32_t opcode = ctx.opcode[(int)instr->opcode];
+#if defined(DEBUG) || !defined(NDEBUG)
+   if (opcode == (uint32_t)-1) {
+      fprintf(stderr, "unsupported opcode: %u - %s\n", (uint32_t) instr->opcode, aco_opcode_string[(size_t) instr->opcode]);
+   }
+#endif // NDEBUG
```

Comment:
what about aco_print_instr()?
maybe something like
```
if (opcode == -1) {
   fprintf(stderr, "Unsupported opcode: ");
   aco_print_instr(instr, stderr);
}
```

Reply:
Hm, that looks good too. I'll look into it.